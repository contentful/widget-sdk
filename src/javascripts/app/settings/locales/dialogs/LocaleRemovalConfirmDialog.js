import React from 'react';
import PropTypes from 'prop-types';
import { TextInput, ModalConfirm } from '@contentful/forma-36-react-components';

export default class LocaleRemovalConfirmDialog extends React.Component {
  static propTypes = {
    isShown: PropTypes.bool.isRequired,
    locale: PropTypes.object.isRequired,
    onCancel: PropTypes.func.isRequired,
    onConfirm: PropTypes.func.isRequired,
  };

  state = {
    typedLocaleCode: '',
  };

  componentDidUpdate(nextProps) {
    if (nextProps.isShown === false) {
      this.setState({ typedLocaleCode: '' });
    }
  }

  render() {
    const { locale } = this.props;
    return (
      <ModalConfirm
        isShown={this.props.isShown}
        onCancel={this.props.onCancel}
        onConfirm={this.props.onConfirm}
        shouldCloseOnOverlayClick={false}
        title={`You're about to delete the ${locale.name} (${locale.code}) locale`}
        intent="negative"
        isConfirmDisabled={this.state.typedLocaleCode !== locale.code}
        confirmLabel="Delete"
        cancelLabel="Don't delete"
        confirmTestId="delete-locale-confirm"
        cancelTestId="delete-locale-cancel">
        <p>
          This will break any API clients that rely on <code>{locale.code}</code> existing.
        </p>
        <p>
          <strong>Please note that this action is permanent and you cannot undo it.</strong>
        </p>
        <p>Please type the locale code below to confirm this change:</p>
        <TextInput
          id="localeInput"
          name="localeInput"
          width="medium"
          testId="repeat-locale-input"
          value={this.state.typedLocaleCode}
          onChange={(e) => {
            this.setState({ typedLocaleCode: e.target.value });
          }}
        />
      </ModalConfirm>
    );
  }
}
