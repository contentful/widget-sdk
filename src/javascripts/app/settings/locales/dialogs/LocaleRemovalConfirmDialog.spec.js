import React from 'react';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';
import LocaleRemovalConfirmDialog from './LocaleRemovalConfirmDialog.es6';

describe('locales/components/LocaleRemovalConfirmDialog', () => {
  const selectors = {
    repeatLocaleInput: 'input[data-test-id="repeat-locale-input"]',
    confirmDeleteLocale: '[data-test-id="delete-locale-confirm"]',
    cancelDeleteLocale: '[data-test-id="delete-locale-cancel"]',
    modalDialogClose: '[data-test-id="modal-dialog-close"]'
  };

  const renderComponent = props => (
    <LocaleRemovalConfirmDialog
      onConfirm={() => {}}
      onCancel={() => {}}
      locale={{
        name: 'English',
        code: 'uk'
      }}
      {...props}
    />
  );

  it('should match snapshot', () => {
    const wrapper = shallow(renderComponent());
    expect(wrapper).toMatchSnapshot();
  });

  it('confirm button should be disabled by default', () => {
    const wrapper = mount(renderComponent());
    expect(wrapper.find(selectors.confirmDeleteLocale)).toBeDisabled();
  });

  it('it is possible to invoke cancel by clicking on two buttons', () => {
    const stubs = {
      onCancel: sinon.stub()
    };
    const wrapper = mount(
      renderComponent({
        onCancel: stubs.onCancel
      })
    );

    wrapper.find(selectors.cancelDeleteLocale).simulate('click');
    wrapper.find(selectors.modalDialogClose).simulate('click');
    expect(stubs.onCancel.callCount).toBe(2);
  });

  it('confirm button should be enabled when user types locale code in input', () => {
    const stubs = {
      onConfirm: sinon.stub(),
      onCancel: sinon.stub()
    };

    const wrapper = mount(
      renderComponent({
        ...stubs
      })
    );
    wrapper.find(selectors.repeatLocaleInput).simulate('change', { target: { value: 'uk' } });
    expect(wrapper.find(selectors.confirmDeleteLocale)).not.toBeDisabled();
    wrapper.find(selectors.confirmDeleteLocale).simulate('click');
    wrapper
      .find(selectors.repeatLocaleInput)
      .simulate('change', { target: { value: 'something' } });
    expect(wrapper.find(selectors.confirmDeleteLocale)).toBeDisabled();
    wrapper.find(selectors.confirmDeleteLocale).simulate('click');

    expect(stubs.onConfirm.callCount).toBe(1);
  });
});
