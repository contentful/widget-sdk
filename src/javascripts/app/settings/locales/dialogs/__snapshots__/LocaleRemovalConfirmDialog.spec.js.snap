// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`locales/components/LocaleRemovalConfirmDialog should match snapshot 1`] = `
<div
  className="modal-dialog"
>
  <header
    className="modal-dialog__header"
  >
    <h1>
      You’re about to delete the
       
      <em>
        English
         (
        uk
        )
      </em>
       
      locale.
    </h1>
    <button
      className="modal-dialog__close"
      data-test-id="modal-dialog-close"
      onClick={[Function]}
    />
  </header>
  <div
    className="modal-dialog__content"
  >
    <div
      className="modal-dialog__richtext"
    >
      <p>
        This will break any API clients that rely on 
        <code>
          uk
        </code>
         existing.
      </p>
      <p>
        <strong>
          Please note that this action is permanent and you cannot undo it.
        </strong>
      </p>
      <p>
        Please type the locale code below to confirm this change:
      </p>
      <TextInput
        disabled={false}
        id="localeInput"
        name="localeInput"
        onChange={[Function]}
        required={false}
        testId="repeat-locale-input"
        value=""
        width="medium"
        withCopyButton={false}
      />
    </div>
  </div>
  <div
    className="modal-dialog__controls"
  >
    <Button
      buttonType="negative"
      disabled={true}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="delete-locale-confirm"
    >
      Delete
    </Button>
    <Button
      buttonType="muted"
      disabled={false}
      indicateDropdown={false}
      isFullWidth={false}
      loading={false}
      onClick={[Function]}
      testId="delete-locale-cancel"
    >
      Don’t delete
    </Button>
  </div>
</div>
`;
